### PR TITLE
Added helpers to collectContentPre and collectContentPost

### PR DESCRIPTION
**Pull request by cristo-rabani**
Function makeContentCollector is used on both sides (client/server) 
but sometimes we need read somethings from html attribute.
and passing data by class is very not elegant way.

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Please read contributing.md for style guide, your indentation looks incorrect here..?

**Comment by cristo-rabani** [comment]
Sorry, I copied and pasted it from my local version.
I used github editor for this for now.
It will be helpful if you prepare editorconfig file ( http://editorconfig.org/ )

In my bitbucket repository I have many core changes,
mostly for optimization like caching in indexedDB for long texts.
Don't serializing data to json for mongo and storing in a few collections instead just in one (This is new database type for ueberDB [I really don't know why someone wants to serializing data to json before save it to mongodb. This slows down loading about ~20%])

Sometimes I change formatting of some files because I hate if '{' is in new line.
This breaks folding of code in webstorm and makes code is too lengthy and is harder to reading for many people but this can be an 'de gustibus'.

**Comment by JohnMcLear** [comment]
Bump @cristo-rabani

**Comment by cristo-rabani** [comment]
check now

File: src/static/js/contentcollector.js [hunk 1]
```diff
@@ -461,7 +468,16 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
     else
     {
       var tname = (dom.nodeTagName(node) || "").toLowerCase();
-
+      var helpers = {
+        //returns attributes like style, id
+        getAttr: function(htmlAttr){
+            return dom.nodeAttr(node, htmlAttr);
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js]: Spacing still wrong here..

File: src/static/js/contentcollector.js [hunk 2]
```diff
@@ -461,7 +468,16 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
...
+        },
+        //gets property of node
+        getProp: function(htmlProp){
+            return dom.nodeProp(node, htmlProp);
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js]: and here..

**Comment by JohnMcLear** [comment]
@cristo-rabani sorry dude there are merge conflicts :!  My bad, that file is active!

Can you please resolve them this is good to merge?

One thing I would like is an example snippit of the helpers being used in a plugin so others can reference this..  I'd also like to see some docs be included, contentCollectorPre is exposed to plugins so the additional functionality needs docs to land in http://etherpad.org/doc/v1.5.0/#index_collectcontentpre

Thanks!

**Comment by JohnMcLear** [comment]
Another bump @cristo-rabani if you want to see this land...

**Comment by JohnMcLear** [comment]
Closing as stale.

**Pull request closed without merging**